components/camel-pg-replication-slot/src/main/java/org/apache/camel/component/pg_replication_slot/PgReplicationSlotConsumer.java

```diff
@@ -0,0 +1,257 @@
+        if (cause != null) {
+            getExceptionHandler().handleException("Error during processing exchange. Will attempt to process the message on next poll.", exchange, cause);
+        }
+    }
+
+    private void createSlot() throws SQLException {
+        this.pgConnection.getReplicationAPI()
+                .createReplicationSlot()
+                .logical()
+                .withSlotName(this.endpoint.getSlot())
+                .withOutputPlugin(this.endpoint.getOutputPlugin())
+                .make();
+    }
+
+    private boolean isSlotCreated() throws SQLException {
+        String sql = String.format("SELECT count(*) FROM pg_replication_slots WHERE slot_name = '%s';", this.endpoint.getSlot());
+
+        try (Statement statement = this.connection.createStatement(); ResultSet resultSet = statement.executeQuery(sql)) {
+            resultSet.next();
+            return resultSet.getInt(1) > 0;
+        }
+    }
+
+    private PGReplicationStream getStream() throws SQLException {
+        if (this.replicationStream != null && !this.replicationStream.isClosed()) {
+            return this.replicationStream;
+        }
+
+        if (isSlotActive()) {
+            log.info(String.format("Slot: %s is active. Waiting for it to be available.", this.endpoint.getSlot()));
+            return null;
+        }
+
+        ChainedLogicalStreamBuilder streamBuilder = this.pgConnection.getReplicationAPI()
+                .replicationStream()
+                .logical()
+                .withSlotName(this.endpoint.getSlot())
+                .withStatusInterval(this.endpoint.getStatusInterval(), TimeUnit.SECONDS);
+
+        Properties slotOptions = new Properties();
+        slotOptions.putAll(this.endpoint.getSlotOptions());
+        streamBuilder.withSlotOptions(slotOptions);
+
+        this.replicationStream = streamBuilder.start();
+
+        return this.replicationStream;
+    }
+
+    private boolean isSlotActive() throws SQLException {
+        String sql = String.format("SELECT count(*) FROM pg_replication_slots where slot_name = '%s' AND active = true;",
+                this.endpoint.getSlot());
+
+        try (Statement statement = this.connection.createStatement(); ResultSet resultSet = statement.executeQuery(sql)) {
+            resultSet.next();
+            return resultSet.getInt(1) > 0;
+        }
+    }
+
+    private void connect() throws SQLException {
+        if (this.connection != null) {
+            this.connection.close();
+        }
+
+        this.connection = this.endpoint.newDbConnection();
+        this.pgConnection = this.connection.unwrap(PGConnection.class);
+        this.replicationStream = null;
+
+        if (this.endpoint.getAutoCreateSlot() && !this.isSlotCreated()) {
+            this.createSlot();
+        }
+    }
+    
+    private void validateConnection(PSQLException exception) {
+        try {
+            if (exception.getCause() instanceof SocketException) {
```

Comment:
for potential reconnect logic, you can take a look the connection reestablishment logics in other components, camel-sql mongo etc. by the looks of it, the first req will fail and reestablish the connection where failing the first request in case of socket exception may be a bit of redundant, and can be improved. this can be of course logged and improved later. not a big worry.

Reply:
I will do that ASAP.

Reply:
Can be improved later on.(As there will be a single connection) No need to mind at the moment.